app/src/main/java/com/github/sdpteam15/polyevents/view/activity/MainActivity.kt

```diff
@@ -74,18 +85,34 @@ class MainActivity : AppCompatActivity() {
             getString(R.string.notification_channel_name)
         )
 
-        //TODO remove to for local cache
-        Settings.mainActivity = this
-        Settings.isLoaded = Settings.isLoaded
-        val intent = Intent(applicationContext, TimerService::class.java)
-        startService(intent)
-        TimerService.instance.observeOnce {
-            it.value.addTask {
-                if (Settings.IsSendingLocationOn)
-                    HelperFunctions.getLoc(this).observeOnce { LatLng ->
-                        if (LatLng.value != null)
-                            currentDatabase.heatmapDatabase!!.setLocation(LatLng.value)
+        localDatabase = (application as PolyEventsApplication).database
+
+        if (TimerService.instance.value == null) {
+            // Get the user settings from local cache and create callback to update the user settings if they changed
+            val userSettingsObservable = ObservableList<UserSettings>()
+            userSettingsObservable.observe(this) {
+                if (it.value.isNotEmpty()) {
+                    userSettingsViewModel.updateUserSettings(
+                        it.value[0]
+                    )
+                }
+            }
+
+            val intent = Intent(applicationContext, TimerService::class.java)
+            startService(intent)
+            TimerService.instance.observeOnce {
+                it.value.addTask {
+                    userSettingsViewModel.getUserSettings(userSettingsObservable)
+                    if (userSettingsObservable.isNotEmpty() && userSettingsObservable[0].isSendingLocationOn) {
+                        HelperFunctions.getLoc(this).observeOnce { LatLng ->
+                            if (LatLng.value != null)
+                                currentDatabase.heatmapDatabase!!.setLocation(
+                                    LatLng.value,
+                                    userSettingsObservable
+                                )
+                        }
                     }
+                }
```

Comment:
Could you add one line of comment to explain what it does

Reply:
Yes will do. But basically this is the same code as before, just with the user settings fetched from local cache.